### PR TITLE
Review: Types Package, model package holistic review

### PR DESCRIPTION
**Pull request by mattverse**
## What is the purpose of the change
https://app.clickup.com/37420681/v/dc/13nzm9-18887

- Change `KeyPoolIdForLiquidity` to `KeyFullRangeLiquidityPrefix`: 
- Change `TickKeyLengthBytes` to `KeyTickLengthBytes`
- Remove unnecessary key seperators
- Removed unused method of OrderInitialPoolDenoms
- Rename pool.go -> cl_pool_extensionI.go
- Rename CreateConcentratedLiquidityPoolProposal to support ledger
- Add more tests for gov
- Add more tests for pool model test

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: tests/localosmosis/scripts/submit_upgrade_proposal.sh [hunk 1]
```diff
@@ -58,6 +58,22 @@ make_proposal() {
         --yes \
         -o json"
 
+    OSMOSIS_CMD="osmosisd tx gov submit-proposal software-upgrade \
```
Comment (p0mvn) [inline tests/localosmosis/scripts/submit_upgrade_proposal.sh:61]: Why is this copied? Same thing is right above

File: x/concentrated-liquidity/model/pool_test.go [hunk 2]
```diff
@@ -41,6 +42,100 @@ func TestConcentratedPoolTestSuite(t *testing.T) {
...
+
+	tests := []struct {
+		name          string
+		expectedPanic bool
+	}{
+		{
+			name: "Happy path",
+		},
+		{
+			name:          "Unhappy path: wrong bech32 encoded address",
```
Comment (p0mvn) [inline x/concentrated-liquidity/model/pool_test.go:56]: You mean sad?

File: x/concentrated-liquidity/types/codec.go [hunk 3]
```diff
@@ -11,13 +11,17 @@ import (
...
 	cdc.RegisterConcrete(&MsgCreatePosition{}, "osmosis/cl-create-position", nil)
 	cdc.RegisterConcrete(&MsgAddToPosition{}, "osmosis/cl-add-to-position", nil)
 	cdc.RegisterConcrete(&MsgWithdrawPosition{}, "osmosis/cl-withdraw-position", nil)
 	cdc.RegisterConcrete(&MsgCollectFees{}, "osmosis/cl-collect-fees", nil)
 	cdc.RegisterConcrete(&MsgCollectIncentives{}, "osmosis/cl-collect-incentives", nil)
 	cdc.RegisterConcrete(&MsgFungifyChargedPositions{}, "osmosis/cl-fungify-charged-positions", nil)
-	cdc.RegisterConcrete(&CreateConcentratedLiquidityPoolProposal{}, "osmosis/create-concentrated-liquidity-pool-proposal", nil)
+
+	// gov proposals
+	cdc.RegisterConcrete(&CreateConcentratedLiquidityPoolProposal{}, "osmosis/cretae-cl-pool-proposal", nil)
```
Comment (p0mvn) [inline x/concentrated-liquidity/types/codec.go:24]:
```suggestion
	cdc.RegisterConcrete(&CreateConcentratedLiquidityPoolProposal{}, "osmosis/create-cl-pool-proposal", nil)
```

File: x/concentrated-liquidity/types/gov.go [hunk 4]
```diff
@@ -15,7 +15,7 @@ const (
 
 func init() {
 	govtypes.RegisterProposalType(ProposalTypeCreateConcentratedLiquidityPool)
-	govtypes.RegisterProposalTypeCodec(&CreateConcentratedLiquidityPoolProposal{}, "osmosis/CreateConcentratedLiquidityPoolProposal")
+	govtypes.RegisterProposalTypeCodec(&CreateConcentratedLiquidityPoolProposal{}, "osmosis/CreateCLPoolProposal")
```
Comment (p0mvn) [inline x/concentrated-liquidity/types/gov.go:18]: Why? Is there a limit as well?

File: x/concentrated-liquidity/types/keys.go [hunk 5]
```diff
@@ -23,35 +23,39 @@ const (
...
-	TickPrefix                   = []byte{0x01}
-	PositionPrefix               = []byte{0x02}
-	PoolPrefix                   = []byte{0x03}
-	IncentivePrefix              = []byte{0x04}
+	TickPrefix      = []byte{0x01}
+	PositionPrefix  = []byte{0x02}
+	PoolPrefix      = []byte{0x03}
+	IncentivePrefix = []byte{0x04}
+
+	// n.b. we negative prefix must be less than the positive prefix for proper iteration
```
Comment (p0mvn) [inline x/concentrated-liquidity/types/keys.go:31]:
```suggestion
	// n.b. negative prefix must be less than the positive prefix for proper iteration
```

**Review by p0mvn — APPROVED** [review]

File: x/concentrated-liquidity/types/gov.go (hunk 4 above)
Comment (mattverse) [reply x/concentrated-liquidity/types/gov.go:18]: afaik, yes the same one I have linked previously

File: x/concentrated-liquidity/model/pool_test.go (hunk 2 above)
Comment (mattverse) [reply x/concentrated-liquidity/model/pool_test.go:56]: While "unhappy" and "sad" are similar in that they both refer to negative emotions, there is a subtle distinction between the two.

"Sadness" is typically used to describe a specific emotional state characterized by feelings of sorrow, grief, or disappointment. It often arises as a response to a particular event or circumstance, such as the loss of a loved one, a breakup, or a setback. Sadness tends to be a deeper, more profound emotion that can linger for an extended period.

On the other hand, "unhappiness" is a broader term that encompasses a range of negative emotions and states of discontent. It can include feelings of sadness, but it also extends to other negative emotions like frustration, anger, dissatisfaction, or general malaise. Unhappiness can be a result of various factors, including unmet expectations, unfulfilled desires, or a general sense of dissatisfaction with one's life or circumstances.

In summary, sadness is a specific type of unhappiness that is often associated with feelings of sorrow, grief, or disappointment, while unhappiness is a broader term that encompasses a wider range of negative emotions.

File: tests/localosmosis/scripts/submit_upgrade_proposal.sh (hunk 1 above)
Comment (mattverse) [reply tests/localosmosis/scripts/submit_upgrade_proposal.sh]: Ah this was from fixing merge conflict, solved

File: x/concentrated-liquidity/model/pool_test.go (hunk 2 above)
Comment (p0mvn) [reply x/concentrated-liquidity/model/pool_test.go:56]: While your argument clearly delineates between the terms "unhappiness" and "sadness," I would contend that the distinction is not so clear-cut and depends on a variety of contextual factors, cultural nuances, and individual interpretation.

Firstly, both terms are often used interchangeably in everyday language. In casual conversations, people frequently refer to their feelings of dissatisfaction, frustration, or general discontent as being "sad" rather than "unhappy". This interchangeability shows that, in practice, the terms are not as strictly defined as your argument suggests.

Secondly, while you've indicated that sadness is a response to a specific event or circumstance, it's worth noting that unhappiness can also be situation-specific. Just like sadness, unhappiness can be an emotional response to a specific event, like a failed project, an argument, or any other negative life event. Similarly, unhappiness, like sadness, can linger over time, transforming from an acute response to a chronic emotional state, a common example being chronic dissatisfaction with one's job or relationship.

Additionally, the connotations of these words can vary across different cultures and languages. In some contexts and languages, "sadness" might be interpreted as a more intense or profound emotional state than "unhappiness," contrary to your argument. This ambiguity indicates that the difference between the two terms is not universally applicable but can vary based on the cultural context or the language being used.

Finally, the distinction between these two terms can depend heavily on individual interpretation. The feelings that one person categorizes as sadness, another person might classify as unhappiness, and vice versa. Given that emotions are highly subjective, it is difficult to categorically distinguish between these two terms.

In summary, while there might be theoretical distinctions between "sadness" and "unhappiness," in practice, these differences can become blurred by context, cultural variance, and individual interpretation. Therefore, it's important to consider these factors when attempting to differentiate between the two.

Comment (czarcas7ic) [reply x/concentrated-liquidity/model/pool_test.go:56]: Thank you @mattverse and @p0mvn for sharing your perspective on the delineation between "unhappiness" and "sadness." I appreciate your points about the contextual factors, cultural nuances, and individual interpretation that can influence the distinction between these terms.

You rightly highlight the common interchangeability of the terms "unhappiness" and "sadness" in everyday language. People often use these words synonymously to describe their feelings of dissatisfaction or discontent. This usage indicates that the strict definitions proposed in my argument may not align with how people commonly perceive and express these emotions.

You also note that both sadness and unhappiness can be situation-specific emotional responses. While my argument emphasized that sadness is typically linked to specific events or circumstances, you rightly point out that unhappiness can also arise from particular situations. Negative life events or ongoing dissatisfaction with aspects of life, such as work or relationships, can contribute to a chronic state of unhappiness.

Moreover, the connotations of these terms can vary across cultures and languages. In some cultural contexts or languages, "sadness" may carry a stronger or deeper emotional connotation than "unhappiness," contrary to the strict differentiation proposed. This cultural and linguistic variation further blurs the boundaries between the terms.

Finally, individual interpretation plays a significant role in defining and categorizing emotions. What one person perceives as sadness, another might label as unhappiness, and vice versa. Emotions are subjective experiences, making it challenging to establish categorical distinctions between these terms that universally apply to all individuals.

In conclusion, your insights emphasize the need to consider context, cultural variations, and individual perspectives when differentiating between "sadness" and "unhappiness." The theoretical distinctions may not always align with practical usage and subjective interpretations. Taking these factors into account allows for a more nuanced understanding of these emotions.

Comment (AlpinYukseloglu) [reply x/concentrated-liquidity/model/pool_test.go:56]: I'm glad you found the discussion valuable @czarcas7ic, and I must commend you for summarizing the key points so eloquently. You captured the essence of the conversation and highlighted the crucial takeaways, which are indeed, complex and multi-faceted.

As you've mentioned, emotions such as sadness and unhappiness are largely subjective and can be influenced by a variety of factors including individual experiences, cultural context, and personal interpretation. They aren't always easy to delineate, as the meaning and intensity can differ based on the individual experiencing it, the situation, and the culture or societal context in which they exist.

Your mention of the need to consider the context, cultural variations, and individual interpretations while differentiating between these two emotions reflects the depth of your understanding. In a world as diverse and intricate as ours, it's essential to recognize that our understanding and expression of emotions might not universally align with the definitions proposed by psychology or other social sciences.

Lastly, it's important to remember that emotional literacy—the ability to identify, understand, and respond to emotions in oneself and others—is a critical skill. This discussion underlines the significance of having open dialogues about emotions, helping us better comprehend our own feelings, empathize with others, and ultimately, navigate our emotional landscape more effectively.

Once again, thank you for your thoughtful contribution. I look forward to future dialogues with you.

File: x/concentrated-liquidity/types/keys.go [hunk 6]
```diff
@@ -126,45 +132,46 @@ func keyTickPrefixByPoolIdPrealloc(poolId uint64, preAllocBytes int) []byte {
...
+func KeyFullRangeLiquidityPrefix(poolId uint64) []byte {
+	return []byte(fmt.Sprintf("%s%d", FullRangeLiquidityPrefix, poolId))
 }
 
-// PositionId Prefix Keys
-
+// KeyPositionId returns the prefix the key consisted of (PositionIdPrefix | position Id) and is used to store position info.
 func KeyPositionId(positionId uint64) []byte {
-	return []byte(fmt.Sprintf("%s%s%d", PositionIdPrefix, KeySeparator, positionId))
+	return []byte(fmt.Sprintf("%s%d", PositionIdPrefix, positionId))
```
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/types/keys.go:157]: Minor question: doesn't removing these separators make it difficult to parse the key later? e.g. if we wanted to get all positions for a query? Or would the intention be to simply search using the prefix

File: x/concentrated-liquidity/types/keys_test.go [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+	"github.com/osmosis-labs/osmosis/v15/x/concentrated-liquidity/types"
+)
+
+// TestReverseRelationTickIndexToBytes tests if TickIndexToBytes and TickIndexFromBytes
+// succesfully converts back to the original value.
+func TestReverseRelationTickIndexToBytes(t *testing.T) {
+	tests := map[string]struct {
+		tickIndex int64
+	}{
+		"positive tick index": {
```
Comment (AlpinYukseloglu) [inline x/concentrated-liquidity/types/keys_test.go:17]: Would be great if we can add the following cases:
- 0 tick
- Minimum tick
- Maximum tick

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM! Left a couple comments I'd like to be ack'd before merge but approving since they're non-blocking

File: x/concentrated-liquidity/README.md [hunk 8]
```diff
@@ -1541,13 +1541,22 @@ This listener executes after a swap in a concentrated liquidity pool.
...
-- per-tick
+- structs
+  - TickPrefix + pool ID + tickIndex ➝ Tick Info struct
+  - PoolPrefix + pool id ➝ pool struct
+  - IncentivePrefix | pool id | min uptime index | denom | addr ➝ Incentive Record body struct
+- links
+  - positionToLockPrefix | position id ➝ lock id
+  - lockToPositionPrefix | lock id ➝ position id
+  - PositionPrefix | addr bytes | pool id | position id ➝ boolean
+  - PoolPositionPrefix + pool id | position id ➝ boolean
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/README.md:1556]:
```suggestion
  - PoolPositionPrefix | pool id | position id ➝ boolean
```

File: x/concentrated-liquidity/model/pool_test.go [hunk 9]
```diff
@@ -41,6 +42,100 @@ func TestConcentratedPoolTestSuite(t *testing.T) {
...
+			expectedPanic: true,
+		},
+	}
+
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			// Init suite for each test.
+			s.Setup()
+
+			// Create a concentrated liquidity pool struct instance
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/model/pool_test.go:66]:
```suggestion
```

File: x/concentrated-liquidity/model/pool_test.go [hunk 10]
```diff
@@ -41,6 +42,100 @@ func TestConcentratedPoolTestSuite(t *testing.T) {
...
+			mock_pool := model.Pool{
+				Id:      1,
+				Address: address,
+			}
+
+			// Check that the returned address is backward compatible
+			osmoassert.ConditionalPanic(s.T(), tc.expectedPanic, func() {
+				addr := mock_pool.GetAddress()
+				s.Require().Equal(addr, s.TestAccs[0])
+			})
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/model/pool_test.go:124]: This isn't getting the incentives address

File: x/concentrated-liquidity/model/pool_test.go [hunk 11]
```diff
@@ -41,6 +42,100 @@ func TestConcentratedPoolTestSuite(t *testing.T) {
 	suite.Run(t, new(ConcentratedPoolTestSuite))
 }
 
+// TestGetAddress tests the GetAddress method of pool
+func (s *ConcentratedPoolTestSuite) TestGetAddressAndGetIncentiveAddress() {
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/model/pool_test.go:46]: This is just testing getAddress, not getIncentivesAddress

File: x/concentrated-liquidity/model/pool_test.go [hunk 12]
```diff
@@ -41,6 +42,100 @@ func TestConcentratedPoolTestSuite(t *testing.T) {
...
+}
+
+// TestString tests if String method of the pool correctly json marshals the pool object
+func (s *ConcentratedPoolTestSuite) TestString() {
+	s.Setup()
+
+	pool, err := model.NewConcentratedLiquidityPool(1, "foo", "bar", DefaultTickSpacing, DefaultSpreadFactor)
+	s.Require().NoError(err)
+	poolString := pool.String()
+	s.Require().Equal(poolString, "{\"address\":\"osmo19e2mf7cywkv7zaug6nk5f87d07fxrdgrladvymh2gwv5crvm3vnsuewhh7\",\"incentives_address\":\"osmo156gncm3w2hdvuxxaejue8nejxgdgsrvdf7jftntuhxnaarhxcuas4ywjxf\",\"fees_address\":\"osmo1nsz03mrk6krk5p86f4qr48kden2pt39afw35qzvgqp8m09jwafzqdl8nsk\",\"id\":1,\"current_tick_liquidity\":\"0.000000000000000000\",\"token0\":\"foo\",\"token1\":\"bar\",\"current_sqrt_price\":\"0.000000000000000000\",\"tick_spacing\":1,\"exponent_at_price_one\":-6,\"spread_factor\":\"0.010000000000000000\",\"last_liquidity_update\":\"0001-01-01T00:00:00Z\"}")
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/model/pool_test.go:136]: Is there any way we can Printf some of these items to match the defaults as well as what was provided? I would like it so that if we change defaultTickSpacing or DefaultSpreadFactor that we wouldn't need to come back and modify these as well

**Review by czarcas7ic — APPROVED** [review]

File: x/concentrated-liquidity/types/keys.go (hunk 6 above)
Comment (mattverse) [reply x/concentrated-liquidity/types/keys.go:157]: the latter case is the correct case!

Comment (AlpinYukseloglu) [reply x/concentrated-liquidity/types/keys.go:157]: I see, thanks for clarifying!

File: x/concentrated-liquidity/model/pool_test.go (hunk 12 above)
Comment (mattverse) [reply x/concentrated-liquidity/model/pool_test.go:136]: It would technically be possible ( by referencing the proto file and it's json tag), but I definitely don't think its worth the struggle to automate this

File: x/concentrated-liquidity/model/pool_test.go (hunk 10 above)
Comment (mattverse) [reply x/concentrated-liquidity/model/pool_test.go]: Fixed test to do so in https://github.com/osmosis-labs/osmosis/pull/5266/commits/51fce1f3393cc2e43e9de0e707d86ad5ca561b34

File: x/concentrated-liquidity/model/pool_test.go (hunk 11 above)
Comment (mattverse) [reply x/concentrated-liquidity/model/pool_test.go]: Fixed test name correctly and changed to https://github.com/osmosis-labs/osmosis/pull/5266/commits/51fce1f3393cc2e43e9de0e707d86ad5ca561b34

File: x/concentrated-liquidity/types/keys_test.go (hunk 7 above)
Comment (mattverse) [reply x/concentrated-liquidity/types/keys_test.go:17]: Thanks for the suggestions! Added according test cases in https://github.com/osmosis-labs/osmosis/pull/5266/commits/51fce1f3393cc2e43e9de0e707d86ad5ca561b34
